### ISSUE TITLE
Decompiler: raise cond-backward-branch loops left flat by StructuringPass (70)

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#914]
## Target: `cond-backward-branch` (70 containers)

`StructuringPass` leaves a container flat on a **conditional backward branch** that should raise to a structured loop (`while`/`do`/`for`) but whose shape the loop-raiser does not yet recognize.

Example flat method: `System.Array.SorterObjectArray::PickPivotAndPartition`

## Background

A fresh CoreLib fidelity sweep on `origin/main` puts the decompiler at **98.09% Full (40227/41012)**. The ~785 Partial methods render as goto-laden flat output because `StructuringPass` leaves their containers flat. `--structuring-stops` tallies *why*:

```
structuring stops over 41012 methods (0 pass bugs):
  10818 containers structured; 1094 left flat across 1065 methods
     472  forward-branch-not-region-exit
     415  cond-target-past-region
     108  unconsumed-regions
      70  cond-backward-branch
      15  eh-terminator-survivor
      14  leave-target-in-container
```

This issue owns **one** of those stop reasons end-to-end, so it can be worked in isolation.

## Reproduce / inspect

```bash
# Per-container histogram of stop reasons (with example methods):
dotnet run --project tools/DecompilerHarness -c Release -- --structuring-stops

# Inspect one method's IR + final C# + cap remarks:
dotnet run --project tools/DecompilerHarness -c Release -- --dump 'Type::Method' --cfg --remarks
```

## Suggested approach (scoped climb — see `docs/decompiler-quality.md`)

1. Run `--structuring-stops`, isolate the **dominant single sub-shape** within this stop reason (don't try to swallow the whole bucket at once).
2. Write the discriminator + **positive/negative adversarial fixtures first** (positive raises in the scorecard; negative pairs differing by one CFG discriminator guard pass/fidelity tests).
3. Raise just that shape; re-measure `--structuring-stops` and the Full count. Land it; repeat for the next sub-shape.

## Acceptance criteria

- A measurable drop in this stop reason's count and a net Full-fidelity gain, with **zero validity regressions** (`--validity-check --diff-validity-defects <baseline>`) and **0 importer bugs**.
- Positive + negative fixtures added.
- Full decompiler test suite green (`dotnet run --project src/ILInspector.Decompiler.Tests -c Release`).

## Coordination note

All CFG structuring stop reasons live in the same `StructuringPass` / `CSharpPrinter` area (a known merge-conflict hotspot — see #754). Design the slice as an additive, focused code path and rebase often; coordinate landing order if sibling structuring issues are in flight.
